### DISCUSSION TITLE
Whats is the best way to deal with protected autenticated routes and public routes in nextjs?

### DISCUSSION THREAD

**Discussion by wilsonneto-swe** [vercel/next.js#18736] [Help] [answered]
Hi guys,
I'm working on a app that we have public and admin routes, in our past CRA app we've used custom routes elements, but we dont have thins in nextjs... We have a lot of public pages, and we have 20 private pages/routes.

Whats is the best way to deal with protected autenticated routes and public routes in nextjs?

Thanks so much!
Best

**Comment by brunocrosier** [comment, marked as answer]
I can recommend https://next-auth.js.org

You can use `getSession()` or `useSession()` to decide what to show on private / protected routes.
